cmake/OpenMP.cmake

```diff
@@ -23,38 +23,22 @@ endif()
 set(OpenMP_cmake_included true)
 include("cmake/Threading.cmake")
 
-if (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+if (APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
It will be useful to add a comment for this update and how it fixes the OpenMP check on MacOS in PR #2051.